src/main/java/at/ac/tuwien/kr/alpha/common/atoms/ExternalLiteral.java

```diff
@@ -114,35 +119,82 @@ public ExternalLiteral substitute(Substitution substitution) {
 	}
 
 	@Override
-	public List<Substitution> getSubstitutions(Substitution partialSubstitution) {
+	public List<Substitution> getSatisfyingSubstitutions(Substitution partialSubstitution) {
 		List<Term> input = getAtom().getInput();
-		List<Term> output = getAtom().getOutput();
-		List<Substitution> substitutions = new ArrayList<>();
 		List<Term> substitutes = new ArrayList<>(input.size());
 
+		// evaluate the external atom
 		for (Term t : input) {
 			substitutes.add(t.substitute(partialSubstitution));
 		}
-
 		Set<List<ConstantTerm<?>>> results = getAtom().getInterpretation().evaluate(substitutes);
-
 		if (results == null) {
 			throw new NullPointerException("Predicate " + getPredicate().getName() + " returned null. It must return a Set.");
 		}
 
-		if (results.isEmpty()) {
-			return emptyList();
+		if (this.isNegated()) {
+			return this.checkNegatedLiteralSatisfied(results) ? Collections.singletonList(partialSubstitution) : Collections.emptyList();
+		} else {
+			return this.buildSubstitutionsForOutputs(partialSubstitution, results);
 		}
+	}
 
-		for (List<ConstantTerm<?>> bindings : results) {
-			if (bindings.size() < output.size()) {
-				throw new RuntimeException("Predicate " + getPredicate().getName() + " returned " + bindings.size() + " terms when at least " + output.size() + " were expected.");
+	/**
+	 * Checks whether this negated external literal is satisfied.
+	 * 
+	 * Note that this method must only be called on negated external literals.
+	 * In that case, the literal itself does not bind any output variables,
+	 * i.e. the underlying atom is satisfied iff the output terms obtained by
+	 * evaluating the underlying external atom match the values to which
+	 * the respective output variables are bound. The result of this method assumes
+	 * that the literal is negated, i.e. for the underlying atom AT, it represents
+	 * the truth value !AT.
+	 * Furthermore, this method assumes that the output terms of the underlying atom
+	 * are {@link ConstantTerm}s, i.e. that the grounder's current partial
+	 * substitution has already been applied to them. That assumption is safe since
+	 * in case of a negated external literal, the output variables are always
+	 * non-binding.
+	 * 
+	 * @param outputs The term lists obtained from evaluating the external atom
```

Comment:
I suggest to rename this parameter, maybe to something like `externalAtomResults` or something else indicating this is not the output of the method.

Reply:
It actually is the output of the method - but renaming definitely makes sense. It's called `externalMethodResult` now.